libs/tx-statemachine/src/transitions/waitCoin.ts

```diff
@@ -14,15 +16,49 @@ export async function waitCoin(container: AnyActionContainer) {
   return receipt
     ? {
         meta: `status: ${receipt.status}`,
+        coin: updateCoinBalance(container, receipt),
       }
     // Our tx has not yet been mined.  While not critical, it is concerning.
     // We have warned, but now return null to allow back-off-and-retry.
     : null
 }
 
+
+function parseLog(contract: TheCoin, log: TransactionReceipt["logs"][0]) {
+  try {
+    return contract.interface.parseLog(log)
+  }
+  catch {
+    return null;
+  }
+}
+export function updateCoinBalance(container: AnyActionContainer, receipt: TransactionReceipt) {
```

Comment:
Instead of just assuming a transaction that was accepted also completed, here we get the actual logs and fetch what was transferred from here

Reply:
Is it saved in the db?

Reply:
in this case, logs means eth logs, (not the same thing as regular logs).

Every action on the chain emits a log, so it's a secure method ot test if something happened.

Reply:
in ethereum, you can only directly query the current state.  If you need historical info, you can only find it in the logs